src/__experimental__/components/tooltip/tooltip.spec.js

```diff
@@ -0,0 +1,359 @@
+import { assertStyleMatch } from '../../../__spec_helper__/test-utils';
+
+function render(props, renderer = TestRenderer.create) {
+  return renderer(<Tooltip { ...props }>Content</Tooltip>);
+}
+
+function renderInner(props) {
+  return TestRenderer.create(<StyledTooltipInner { ...props } />).toJSON();
+}
+
+function renderPointer(props) {
+  return TestRenderer.create(<StyledTooltipPointer { ...props } />).toJSON();
+}
+
+function renderWrapper(props) {
+  return TestRenderer.create(<StyledTooltipWrapper { ...props } />).toJSON();
+}
+
+describe('Tooltip', () => {
+  describe('default props', () => {
+    it('matches snapshot', () => {
+      const wrapper = render({}, mount);
+
+      expect(wrapper).toMatchSnapshot();
+    });
+  });
+
+  describe('visible', () => {
+    it('matches snapshot', () => {
+      const wrapper = render({ isVisible: true }).toJSON();
+
+      expect(wrapper).toMatchSnapshot();
+    });
+  });
+
+  describe('styles', () => {
+    describe('TooltipInner', () => {
+      it('default', () => {
+        assertStyleMatch(
+          {
+            backgroundColor: '#00141D',
+            color: '#FFFFFF',
+            display: 'inline-block',
+            fontWeight: '700',
+            padding: '10px 15px',
+            textAlign: 'center',
+            maxWidth: '300px',
+            wordBreak: 'normal',
+            whiteSpace: 'pre-wrap'
+          },
+          renderInner()
+        );
+      });
+
+      it('type="error"', () => {
+        assertStyleMatch(
+          { backgroundColor: '#C7384F' }, renderInner({ type: 'error' })
+        );
+      });
+    });
+
+    describe('TooltipWrapper', () => {
+      it('default', () => {
+        assertStyleMatch(
+          {
+            position: 'absolute',
+            width: '300px',
+            zIndex: '1003'
+          },
+          renderWrapper(),
+        );
+      });
+
+      it('position === "bottom"', () => {
+        assertStyleMatch(
+          { textAlign: 'center' },
+          renderWrapper({ position: 'bottom' })
+        );
+      });
+
+      it('position === "top"', () => {
+        assertStyleMatch(
+          { textAlign: 'center' },
+          renderWrapper({ position: 'top' })
+        );
+      });
+
+      it('position === "left"', () => {
+        assertStyleMatch(
+          { textAlign: 'left' },
+          renderWrapper({ position: 'left' })
+        );
+      });
+
+      it('position === "right"', () => {
+        assertStyleMatch(
+          { textAlign: 'right' },
+          renderWrapper({ position: 'right' })
+        );
+      });
+
+      it('align === "left"', () => {
+        assertStyleMatch(
+          { textAlign: 'left' },
+          renderWrapper({ align: 'left' })
+        );
+      });
+
+      it('align === "right"', () => {
+        assertStyleMatch(
+          { textAlign: 'right' },
+          renderWrapper({ align: 'right' })
+        );
+      });
```

Comment:
you could use `describe.each()` for position and alignment separately.

Reply:
Jest library has its own method for creating loops:
```
describe.each(OptionsHelper.alignBinary)('for the position prop set to "%s"', (align) => {
  it('should have it's text aligned to ${align}', () =>
.
.
.
})
```

Reply:
Inclined to agree, using describe.each is a nicer syntax and more readable